### PR TITLE
Us crop stats

### PR DESCRIPTION
**Pull request by poudelpratishtha**
Added an R script to access NASS data.

### CODE REVIEW

**Review by krsnapaudel — CHANGES_REQUESTED** [review]
Is it possible to make the script work for other crops? This is what I tried for wheat.
`
nass_data_items <- c("ACRES HARVESTED",
                                   "YIELD, MEASURED IN BU / ACRE",
                                   "PRODUCTION MEASURED IN BU")

if (crop == "CORN") {
      data_item <- paste(crop, ", GRAIN - ", nass_data_items[i], sep="")
} else {
       data_item <- paste(crop, " - ", nass_data_items[i], sep="")
}
`
Maybe there is a clean way.
Another way is to add comments describing what you need to do if you change the crop.
Then the old script can be removed and the new script can be renamed to `nass_stats.R`.

**Review by krsnapaudel — COMMENTED** [review]
Also `nass_stats_file` is not defined.

**Comment by krsnapaudel** [comment]
I see a few issues:
* Only AREA HARVESTED is exported. I don't see the values of other categories in the file that is written.
* Note the column `value` must be renamed to the category that we requested. When category is AREA_HARVESTED, rename value to AREA_HARVESTED. Similar for YIELD and PRODUCTION.
* The dataframes for three categories must be merged so value for each category is appended as a column.
* The dataframes for different years must be concatenated vertically (with `rbind`).

Here is some code I wrote. You could make it cleaner:

```
nass_stats <- NULL
for (crop in nass_crops) {
  nass_crop_stats <- NULL
  for (yr in years) {
    nass_yr_stats <- NULL
    for (i in seq_along(nass_categories)) {
      category <- nass_categories[i]
      if (crop == "CORN") {
        data_item <- paste(crop, ", GRAIN - ", nass_data_items[i], sep="")
      } else {
        data_item <- paste(crop, " - ", nass_data_items[i], sep="")
      }

      print(data_item)
      query_result <- tidyUSDA::getQuickstat(
        program ="SURVEY",
        sector="CROPS",
        group="FIELD CROPS",
        commodity=crop,
        category=category,
        data_item = data_item,
        domain="TOTAL",
        geographic_level="COUNTY",
        key = api_key,
        year = as.character(yr))

      query_result <- query_result[,c('county_code', 'county_name', 'state_name', 'year', 'Value')]
      names(query_result)[names(query_result) == 'Value'] <- category
      query_result['crop_name'] <- crop

      # merge data for different categories
      if (is.null(nass_yr_stats)) {
        nass_yr_stats <- query_result
      } else {
        nass_yr_stats <- merge(nass_yr_stats, query_result,
                               by=c('crop_name', 'county_code', 'county_name', 'state_name', 'year'))
      }
    }

    # combine data for different years
    if (is.null(nass_crop_stats)) {
      nass_crop_stats <- nass_yr_stats
    } else {
      nass_crop_stats <- rbind(nass_crop_stats, nass_yr_stats)
    }
  }

  # combine data for different crops
  if (is.null(nass_stats)) {
    nass_stats <- nass_crop_stats
  } else {
    nass_stats <- rbind(nass_stats, nass_crop_stats)
  }
}
```

File: data_preparation/crop_statistics_US/nass_corn_stats.R
Comment (poudelpratishtha) [inline data_preparation/crop_statistics_US/nass_corn_stats.R:1]: I have added the script to include winter wheat and grain corn. It should be easily modifiable to add more crops if needed. Let me know if it doesn't work on your end; the resulting data should have columns: Year, Crop, State, County, Area, Yield, and Production for corn and wheat. We should document the type of crops i.e. grain corn and winter wheat and units of the variables in ReadME. Have you already set up a ReadME?

Comment (krsnapaudel) [reply data_preparation/crop_statistics_US/nass_corn_stats.R:1]: Yes thank you.

Comment (krsnapaudel) [reply data_preparation/crop_statistics_US/nass_corn_stats.R:1]: The script looks good. I will test once on Friday and we can merge.

Comment (poudelpratishtha) [reply data_preparation/crop_statistics_US/nass_corn_stats.R:1]: In this line, raw_data <- map_dfr(get_items, nassqs), it should be get_params not get_items.

**Comment by krsnapaudel** [comment]
@poudelpratishtha : Could you remove old `nass_stats.R`? and update the data cards based on [new data card template](https://docs.google.com/document/d/1DLuWiGuwlK__qcbmUDhcMjPtwNaTCX6zSPCcvBPuto0/edit?usp=sharing)? This is otherwise ready to merge. Thanks.

**Review by krsnapaudel — CHANGES_REQUESTED** [review]
The shapefiles data card needs update based on the new template.

**Pull request closed without merging**